internal/status/handlers/nvidia.go

```diff
@@ -0,0 +1,351 @@
+		MemClock      string `xml:"mem_clock"`
+	} `xml:"default_applications_clocks"`
+	MaxClocks struct {
+		GraphicsClock string `xml:"graphics_clock"`
+		SmClock       string `xml:"sm_clock"`
+		MemClock      string `xml:"mem_clock"`
+		VideoClock    string `xml:"video_clock"`
+	} `xml:"max_clocks"`
+	MaxCustomerBoostClocks struct {
+		GraphicsClock string `xml:"graphics_clock"`
+	} `xml:"max_customer_boost_clocks"`
+	ClockPolicy struct {
+		AutoBoost        string `xml:"auto_boost"`
+		AutoBoostDefault string `xml:"auto_boost_default"`
+	} `xml:"clock_policy"`
+	Voltage struct {
+		GraphicsVolt string `xml:"graphics_volt"`
+	} `xml:"voltage"`
+	SupportedClocks struct {
+		SupportedMemClock []struct {
+			Value                  string   `xml:"value"`
+			SupportedGraphicsClock []string `xml:"supported_graphics_clock"`
+		} `xml:"supported_mem_clock"`
+	} `xml:"supported_clocks"`
+	// TODO: `Processes` is wrong, need to accomodate running processes
+	Processes          processes `xml:"processes"`
+	AccountedProcesses string    `xml:"accounted_processes"`
+}
+
+type processes struct {
+	XMLName     xml.Name `xml:"processes"`
+	Text        string   `xml:",chardata"`
+	ProcessInfo []struct {
+		Text              string `xml:",chardata"`
+		GpuInstanceID     string `xml:"gpu_instance_id"`
+		ComputeInstanceID string `xml:"compute_instance_id"`
+		Pid               string `xml:"pid"`
+		Type              string `xml:"type"`
+		ProcessName       string `xml:"process_name"`
+		UsedMemory        string `xml:"used_memory"`
+	} `xml:"process_info"`
+}
+
+// Filter down the relevant information from our nvidia-smi dump
+func (xml NvidiaSmiLog) FilterStatus() (status.GPUStatusPacket, error) {
+	fanSpeed, err := parseUIntWithUnit(xml.Gpu.FanSpeed)
+	isDirty := err != nil
+	memoryTotal, err := parseUIntWithUnit(xml.Gpu.FbMemoryUsage.Total)
+	isDirty = isDirty || err != nil
+	memoryUsed, err := parseUIntWithUnit(xml.Gpu.FbMemoryUsage.Used)
+	isDirty = isDirty || err != nil
+	temp, err := parseIntWithUnit(xml.Gpu.Temperature.GpuTemp)
+	isDirty = isDirty || err != nil
+	gpuUtil, err := parseUIntWithUnit(xml.Gpu.Utilization.GpuUtil)
+	isDirty = isDirty || err != nil
+	memUtil, err := parseUIntWithUnit(xml.Gpu.Utilization.MemoryUtil)
+	isDirty = isDirty || err != nil
+
+	res := status.GPUStatusPacket{
+		Name:              xml.Gpu.ProductName,
+		Brand:             xml.Gpu.ProductBrand,
+		DriverVersion:     xml.DriverVersion,
+		MemoryTotal:       memoryTotal,
+		MemoryUtilisation: memUtil,
+		GPUUtilisation:    gpuUtil,
+		MemoryUsed:        memoryUsed,
+		FanSpeed:          fanSpeed,
+		Temp:              temp}
+
+	// report back catastrophic failures
+	if isDirty {
+		return res, errors.New("Parsing of data fields failed.")
+	}
+
+	return res, nil
+}
+
+// Helper function for extracting GPU data such as available memory and fan speed
+func parseUIntWithUnit(input string) (uint64, error) {
+	parts := strings.Split(input, " ")
```

Comment:
It looks like we're just discarding the unit here, which feels bad.

If we genuinly don't need it, it's nicer to use [strings.Cut](https://pkg.go.dev/strings#Cut), although that might be nicer anyway